src/test/scala/mesosphere/marathon/api/akkahttp/v2/TasksControllerTest.scala

```diff
@@ -161,6 +162,63 @@ class TasksControllerTest extends UnitTest with ScalatestRouteTest with Inside w
     }
   }
 
+  "List tasks as txt" should {
+    "list (txt) tasks with less ports than the current app version" in new Fixture {
+      // Regression test for #234
+      Given("one app with one task with less ports than required")
+      val app = AppDefinition("/foo".toRootPath, portDefinitions = Seq(PortDefinition(0), PortDefinition(0)))
+      val instance = TestInstanceBuilder.newBuilder(app.id).addTaskRunning().getInstance()
+
+      val tasksByApp = InstanceTracker.InstancesBySpec.forInstances(instance)
+      instanceTracker.instancesBySpec returns Future.successful(tasksByApp)
+
+      val rootGroup = createRootGroup(apps = Map(app.id -> app))
+      groupManager.rootGroup() returns rootGroup
+
+      assert(app.servicePorts.size > instance.appTask.status.networkInfo.hostPorts.size)
+
+      When("Getting the txt tasks index")
+      Get(Uri./).addHeader(Accept(MediaTypes.`text/plain`)) ~> controller.route ~> check {
+        Then("The status should be 200")
+        status should be(StatusCodes.OK)
+
+        responseAs[String] should include (s"${app.id.safePath}\t${app.servicePorts.head}")
```

Comment:
Could we be explicit here? I think that's easier to understand.

Reply:
what do you mean by being explicit?